frontend/src/lib/components/TaxonomicFilter/infiniteListLogic.ts

```diff
@@ -240,11 +240,33 @@ export const infiniteListLogic = kea<infiniteListLogicType>([
         hasRemoteDataSource: [(s) => [s.remoteEndpoint], (remoteEndpoint) => !!remoteEndpoint],
         rawLocalItems: [
             (selectors) => [
-                (state, props) => {
+                (state, props: InfiniteListLogicProps) => {
                     const taxonomicGroups = selectors.taxonomicGroups(state)
                     const group = taxonomicGroups.find((g) => g.type === props.listGroupType)
+
+                    const isCohortArray = (items: any): items is CohortType[] => {
+                        return Array.isArray(items) && items.every((item) => 'filters' in item)
+                    }
+
+                    const filterNonBehavioralCohorts = (items: CohortType[]): CohortType[] => {
+                        return items.filter((item: CohortType) => {
+                            if (item.filters?.properties?.values) {
+                                return !item.filters.properties.values.some((value: any) =>
+                                    value.values?.some((subValue: any) => subValue.type === 'behavioral')
```

Comment:
flyby this isn't really enough, if you have a cohort, that depends on another cohort, and this second cohort is behavioural the API will still reject it 😅

Reply:
ooooh good catch, I didn't think of that edge case.  Thanks for the comment!

Reply:
fixed this and updated my demo, thanks for the callout @neilkakkar